api/openapi/model-registry.yaml

```
@@ -0,0 +1,1713 @@
+        -
+          $ref: '#/components/schemas/BaseResource'
+        -
+          required:
+            - artifactType
+          type: object
+          properties:
+            artifactType:
+              type: string
+    ServingEnvironmentList:
+      description: List of ServingEnvironments.
+      type: object
+      allOf:
+        -
+          type: object
+          properties:
+            items:
+              description: ''
+              type: array
+              items:
+                $ref: '#/components/schemas/ServingEnvironment'
+              readOnly: false
+        -
+          $ref: '#/components/schemas/BaseResourceList'
+    RegisteredModelList:
+      description: List of RegisteredModels.
+      type: object
+      allOf:
+        -
+          type: object
+          properties:
+            items:
+              description: ''
+              type: array
+              items:
+                $ref: '#/components/schemas/RegisteredModel'
+              readOnly: false
+        -
+          $ref: '#/components/schemas/BaseResourceList'
+    ServingEnvironment:
+      description: A Model Serving environment for serving `RegisteredModels`.
+      allOf:
+        -
+          $ref: '#/components/schemas/BaseResource'
+        -
+          type: object
+        -
+          $ref: '#/components/schemas/ServingEnvironmentCreate'
+    ServingEnvironmentUpdate:
+      description: A Model Serving environment for serving `RegisteredModels`.
+      allOf:
+        -
+          $ref: '#/components/schemas/BaseResourceUpdate'
+    ServingEnvironmentCreate:
+      description: A Model Serving environment for serving `RegisteredModels`.
+      allOf:
+        -
+          type: object
+        -
+          $ref: '#/components/schemas/BaseResourceCreate'
+        -
+          $ref: '#/components/schemas/ServingEnvironmentUpdate'
+    InferenceService:
+      description: >-
+        An `InferenceService` entity in a `ServingEnvironment` represents a deployed `ModelVersion`
+        from a `RegisteredModel` created by Model Serving.
+      allOf:
+        -
+          $ref: '#/components/schemas/BaseResource'
+        -
+          $ref: '#/components/schemas/InferenceServiceCreate'
+    InferenceServiceCreate:
+      description: >-
+        An `InferenceService` entity in a `ServingEnvironment` represents a deployed `ModelVersion`
+        from a `RegisteredModel` created by Model Serving.
+      required:
+        - registeredModelId
+        - servingEnvironmentId
+      allOf:
+        -
+          $ref: '#/components/schemas/BaseResourceCreate'
+        -
+          $ref: '#/components/schemas/InferenceServiceUpdate'
+      properties:
+        registeredModelId:
+          description: ID of the `RegisteredModel` to serve.
+          type: string
+        servingEnvironmentId:
+          description: ID of the parent `ServingEnvironment` for this `InferenceService` entity.
+          type: string
```

not necessary in this PR (since is for _Serving_) but here I believe is missing the "anonymous object" ? 🤔 

```suggestion
      allOf:
        -
          $ref: '#/components/schemas/BaseResourceCreate'
        -
          $ref: '#/components/schemas/InferenceServiceUpdate'
        -
          type: object
          properties:
            registeredModelId:
              description: ID of the `RegisteredModel` to serve.
              type: string
            servingEnvironmentId:
              description: ID of the parent `ServingEnvironment` for this `InferenceService` entity.
              type: string
```

---

Fixed.

---

👍